### PR TITLE
Page Pattern: Fix site title, tagline and logo in the preview

### PR DESCRIPTION
**Pull request by arthur791004**
#### Proposed Changes

* Switch to the `block-previews/pattern` endpoint
* Pass `wpcom_site` to that endpoint and the site-related block will based on the site we passed

#### Testing Instructions

* Proxy <your_site> and widgets.wp.com to your sandbox
* run `cd apps/editing-toolkit && yarn dev --sync`. Or run `install-plugin.sh editing-toolkit fix/page-pattern-modal-preview` on your sandbox.
* Go to "Pages > Add New"
* When you see the page pattern modal, select "Link in bio" category
* Check the site title, tagline and logo are rendered correctly

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to p1666257227083559-slack-CRWCHQGUB

### CODE REVIEW

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts [hunk 1]
```diff
@@ -14,7 +14,9 @@ let handled = false;
 domReady( () => {
 	// If site launch options does not exist, stop.
 	const siteLaunchOptions = window.wpcomEditorSiteLaunch;
-	if ( ! siteLaunchOptions ) return;
+	if ( ! siteLaunchOptions ) {
```
Comment (arthur791004) [inline apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts:17]: Not sure why it's auto-formatted

File: apps/editing-toolkit/editing-toolkit-plugin/global-styles/src/global-styles-sidebar.js [hunk 2]
```diff
@@ -37,7 +37,7 @@ const PanelActionButtons = ( {
 			{ __( 'Reset', 'full-site-editing' ) }
 		</Button>
 		<Button
-			className={ 'global-styles-sidebar__publish-button' }
+			className="global-styles-sidebar__publish-button"
```
Comment (arthur791004) [inline apps/editing-toolkit/editing-toolkit-plugin/global-styles/src/global-styles-sidebar.js:40]: Not sure why it's auto-formatted either...

**Review by taipeicoder — APPROVED** [review]
LGTM! 👍

File: packages/page-pattern-modal/src/components/pattern-selector-item.tsx [hunk 3]
```diff
@@ -6,25 +6,39 @@ interface PatternSelectorItemProps {
...
+
+const PATTERN_SOURCE_SITE = 'dotcompatterns.wordpress.com';
+
+const getPatternPreviewUrl = (
+	stylesheet: string,
+	language: string,
+	patternId: string
+): string => {
+	const params = new URLSearchParams( {
+		wpcom_site: window._currentSiteId.toString(),
```
Comment (miksansegundo) [inline packages/page-pattern-modal/src/components/pattern-selector-item.tsx]: It should be `demo_site`, not `wpcom_site`.

Comment (arthur791004) [reply packages/page-pattern-modal/src/components/pattern-selector-item.tsx]: yes, I'll update it

**Review by miksansegundo — COMMENTED** [review]
Just a reminder to use the param name `demo_site` to match the latest changes in the diff.

File: packages/page-pattern-modal/src/components/pattern-selector-item.tsx [hunk 4]
```diff
@@ -6,25 +6,39 @@ interface PatternSelectorItemProps {
 	onSelect: ( value: string ) => void;
 	patternPostID: number | null;
 	title?: string;
+	stylesheet: string;
```
Comment (fushar) [inline packages/page-pattern-modal/src/components/pattern-selector-item.tsx:9]: Can we then remove the `theme` props everywhere?

Comment (arthur791004) [reply packages/page-pattern-modal/src/components/pattern-selector-item.tsx:9]: The API still keeps using the stylesheet from query parameter, so we cannot remove  it now.

Comment (fushar) [reply packages/page-pattern-modal/src/components/pattern-selector-item.tsx:9]: No I mean `theme` prop. For example it is not used anymore this file.

Comment (arthur791004) [reply packages/page-pattern-modal/src/components/pattern-selector-item.tsx:9]: Oops...I didn't notice we have that prop 🙈

File: packages/page-pattern-modal/src/components/page-pattern-modal.tsx [hunk 5]
```diff
@@ -26,6 +26,7 @@ interface PagePatternModalProps {
 	onClose: () => void;
 	siteInformation?: Record< string, string >;
 	patterns: PatternDefinition[];
+	stylesheet?: string;
```
Comment (fushar) [inline packages/page-pattern-modal/src/components/page-pattern-modal.tsx:29]: Also `theme` here and everywhere else 🙈

Comment (arthur791004) [reply packages/page-pattern-modal/src/components/page-pattern-modal.tsx:29]: Removed

**Comment by fushar** [comment]
I get this top banner, is this expected? 
image

**Comment by fushar** [comment]
image

**Comment by arthur791004** [comment]
I guess your sandbox is not outdated 😅

**Comment by fushar** [comment]
No, it is not.

FYI, the test site is: https://testfushar36.wordpress.com/

**Comment by arthur791004** [comment]
Fixed it via D90903-code, nice caught!

**Review by fushar — APPROVED** [review]
Looks good now!

File: apps/editing-toolkit/editing-toolkit-plugin/starter-page-templates/class-starter-page-templates.php [hunk 6]
```diff
@@ -184,7 +184,7 @@ public function enqueue_assets() {
 				'templates'    => array_merge( $default_templates, $page_templates ),
 				// phpcs:ignore WordPress.Security.NonceVerification.Recommended
 				'screenAction' => isset( $_GET['new-homepage'] ) ? 'add' : $screen->action,
-				'theme'        => normalize_theme_slug( get_stylesheet() ),
+				'stylesheet'   => get_stylesheet(),
```
Comment (arthur791004) [inline apps/editing-toolkit/editing-toolkit-plugin/starter-page-templates/class-starter-page-templates.php:187]: Not sure why the returned value of `get_stylesheet()` doesn't have `repo` prefix such as `pub/`, `premium`, etc. Prepare to revert this change: https://github.com/Automattic/wp-calypso/pull/69553

**Comment by arthur791004** [comment]
Not sure why the returned value of `get_stylesheet()` doesn't have `repo` prefix such as `pub/`, `premium/`, etc on AT sites. Prepare to revert this change: https://github.com/Automattic/wp-calypso/pull/69553
